source/extensions/common/dynamic_forward_proxy/dns_cache_impl.h

```diff
@@ -270,6 +270,7 @@ class DnsCacheImpl : public DnsCache, Logger::Loggable<Logger::Id::forward_proxy
   absl::Mutex ip_version_to_remove_lock_;
   absl::optional<Network::Address::IpVersion>
       ip_version_to_remove_ ABSL_GUARDED_BY(ip_version_to_remove_lock_) = absl::nullopt;
+  bool enable_dfp_dns_trace_;
```

Comment:
if we think we want the traces long term I think we should comment a bit more about trace lifetime somewhere